python/mlc_chat/compiler/quantization/ft_quantization.py

```diff
@@ -0,0 +1,399 @@
+        model: nn.Module,
+        quant_map: QuantizeMapping,
+        name_prefix: str,
+    ) -> nn.Module:
+        """
+        Quantize model with FasterTransformer quantization
+
+        Parameters
+        ----------
+        model : nn.Module
+            The non-quantized nn.Module.
+
+        quant_map : QuantizeMapping
+            The quantize mapping with name mapping and func mapping.
+
+        name_prefix : str
+            The name prefix for visited weight.
+
+        Returns
+        -------
+        ret : nn.Module
+            The quantized nn.Module.
+        """
+
+        class _Mutator(nn.Mutator):
+            def __init__(self, config: FTQuantize, quant_map: QuantizeMapping) -> None:
+                super().__init__()
+                self.config = config
+                self.quant_map = quant_map
+
+            def visit_module(self, name: str, node: nn.Module) -> Any:
+                """
+                The visiting method for FasterTransformer quantization of nn.Module nodes.
+
+                Parameters
+                ----------
+                name : str
+                    The name of the current node.
+
+                node : nn.Module
+                    The current node of nn.Module to mutate.
+
+                Returns
+                ------
+                ret_node: Any
+                    The new node to replace current node.
+                """
+                if isinstance(node, nn.Linear):
+                    weight_name = f"{name}.weight"
+                    self.quant_map.param_map[weight_name] = [f"{name}.q_weight", f"{name}.q_scale"]
+                    self.quant_map.map_func[weight_name] = self.config.quantize_weight
+                    return FTQuantizeLinear.from_linear(node, self.config)
+                if isinstance(node, nn.Embedding):
+                    weight_name = f"{name}.weight"
+                    self.quant_map.param_map[weight_name] = [f"{name}.q_weight", f"{name}.q_scale"]
+                    group_quantize = self.config.fallback_group_quantize()
+                    self.quant_map.map_func[weight_name] = group_quantize.quantize_weight
+                    return GroupQuantizeEmbedding.from_embedding(node, group_quantize)
+                return self.visit(name, node)
+
+        model.to(dtype=self.model_dtype)
+        mutator = _Mutator(self, quant_map)
+        model = mutator.visit(name_prefix, model)
+        return model
+
+    def quantize_weight(self, weight: NDArray) -> List[NDArray]:
+        """
+        Quantize weight with FasterTransformer quantization
+
+        Parameters
+        ----------
+        weight : NDArray
+            The original weight.
+
+        Returns
+        ------
+        ret: List[NDArray]
+            The list of FasterTransformer quantized weights.
+        """
+        assert tvm.get_global_func(
+            "relax.ext.cutlass", True
+        ), "Cutlass should be enabled in runtime, to quantize weight"
```

Comment:
Update the error message to be more indicative: https://github.com/mlc-ai/mlc-llm/pull/1480#discussion_r1436642211